Content.Server/Mousetrap/MousetrapSystem.cs

```diff
@@ -0,0 +1,98 @@
+using Content.Shared.Damage;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Inventory;
+using Content.Shared.Mousetrap;
+using Content.Shared.StepTrigger;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.Mousetrap;
+
+public sealed class MousetrapSystem : EntitySystem
+{
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MousetrapComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<MousetrapComponent, StepTriggerAttemptEvent>(OnStepTriggerAttempt);
+        SubscribeLocalEvent<MousetrapComponent, StepTriggeredEvent>(OnStepTrigger);
+    }
+
+    private void Trigger(EntityUid uid, MousetrapComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+        {
+            return;
+        }
+
+        SoundSystem.Play(Filter.Pvs(uid), component.SoundOnActivate.GetSound());
+
+        component.IsActive = false;
+    }
+
+    private void OnUseInHand(EntityUid uid, MousetrapComponent component, UseInHandEvent args)
+    {
+        component.IsActive = !component.IsActive;
+
+        UpdateVisuals(uid);
+    }
+
+    private void OnStepTriggerAttempt(EntityUid uid, MousetrapComponent component, ref StepTriggerAttemptEvent args)
```

Comment:
why is this occurring on step trigger attempt? i assume its for ordering purposes so that DamageOnTrigger doesn't apply, but I don't see why they can't just both occur on `StepTriggeredEvent`

Reply:
the duplication here with `DamageOnTrigger` is a little weird too, feels like the `DamageOnTrigger` should be raising an event to modify damage that `Mousetrap` then subscribes to, like how melee weapons work with wielding

Reply:
i had some odd ordering issues related to this, but it's probably better to do something like `BeforeDamageOnTrigger` so that the damage could be modified, most likely

Reply:
yeah, that is what should be done imo